docs/compactor.md

```diff
@@ -0,0 +1,93 @@
+
+**Proposed Directory Structure**
+```yaml
+Backup :
+    - Full-Snapshot-revision-1
+    - Incremental-Snapshot-revision-6-1 (Format- Incremental-Snapshot-revision-<Revision Number After Last Operation>-<Revision Number Before First Operation>)
+    - Incremental-Snapshot-revision-12-6
+    - Incremental-Snapshot-revision-14-12
+    - Incremental-Snapshot-revision-18-14
+    - Full-Snapshot-revision-18 (Compacted)
+    - Incremental-Snapshot-revision-22-18
+    - Incremental-Snapshot-revision-27-22
+    - Incremental-Snapshot-revision-32-27
+    - Incremental-Snapshot-revision-38-32
+    - Full-Snapshot-revision-38 (Compacted)
+    - Incremental-Snapshot-revision-41-38
+    - Incremental-Snapshot-revision-43-41
+    - Incremental-Snapshot-revision-48-43
+    - Incremental-Snapshot-revision-52-48
+```
+
+3. A meta-data file might need to be added in the Backup directory to mark which compacted (or full) snapshot is ready to be used as the base for restoration. Restorer would study the meta-data file and pickup the ready full snapshot from there and apply all the delta snapshots having revision number greater than that. So, with the backup directory as above, if the content of metadata file is like following:
+
+* Full-Snapshot-revision-1
+* Full-Snapshot-revision-18
+
+Then, during restoration Full-Snapshot-revision-18, and all the delta snaphots from Incremental-Snapshot-revision-22-18 to Incremental-Snapshot-revision-52-48 will be applied. **Restorer would restore from latest ready full snapshot and all subsequent delta snapshots.** As Full-Snapshot-revision-38 is not present in the metadata file, it would not be considered as ready yet.
+
+Another approach to make sure that restorer only picks up snapshots which are ready, full snapshots can be named with extension `.part` until the snapshot is ready. When the snapshot is ready, `.part` extension can be removed from it's name. Restorer would avoid any file having extension as `.part`.  
+
+So, if Full-Snapshot-revision-38 is not ready in the above example, it can be named as `Full-Snapshot-revision-38.part`. Restorer will apply Full-Snapshot-revision-18 and all the delta snaphots from Incremental-Snapshot-revision-22-18 to Incremental-Snapshot-revision-52-48. **Restorer would restore from latest ready full snapshot and all subsequent delta snapshots.**
+
+This option is preferred as it avoids introducing a metadata file. But it would work only if renaming to remove the `.part` extension (upon completion of upload) is an atomic operation in all the supported snapstores. This needs to be evaluated.
+
+#### Backward Compatibility
+1. **Restoration** : The changes to handle the newly proposed backup directory structure must be backward compatible with older structures at least for restoration because we need have to restore from backups in the older structure. This includes the support for restoring from a backup without a metadata file if that is used in the actual implementation.
```

Comment:
Can you please remove the reference to the metadata file and mention all the three cases for restoration (latest snapshots in v1 only, in v2 only and in both v1 and v2)?